File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/comment/direct/Java/Python/s893190600_commented.py
Here's the Python code based on the Java code provided:


import sys

class MyInput:
    def __init__(self):
        self.tokens = None
        self.f = sys.stdin

    def get_token(self):
        while self.tokens is None or len(self.tokens) == 0:
            line = self.f.readline()
            if not line:
                return None
            self.tokens = line.strip().split()
        return self.tokens.pop(0)

    def nextInt(self):
        return int(self.get_token())

    def nextLong(self):
        return int(self.get_token())

    def nextString(self):
        return self.get_token()

    def nextDouble(self):
        return float(self.get_token())

class PP:
    def __init__(self, key, val):
        self.key = key
        self.val = val

class PPP:
    def __init__(self, key, val1, val2):
        self.key = key
        self.val1 = val1
        self.val2 = val2

class PPL:
    def __init__(self, key, val):
        self.key = key
        self.val = val

class PPDL:
    def __init__(self, key, val):
        self.key = key
        self.val = val

    def dump(self, out):
        out.write(f"key = {self.key}  val ")
        for v in self.val:
            out.write(f"[{v}] ")
        out.write("\n")

class PPKEY:
    def __init__(self, key, val):
        self.key = key
        self.val = val

    def __eq__(self, other):
        if isinstance(other, PPKEY):
            return self.key == other.key and self.val == other.val
        return False

    def __hash__(self):
        return hash((self.key, self.val))

class PPLKEY:
    def __init__(self, key, val):
        self.key = key
        self.val = val

    def __eq__(self, other):
        if isinstance(other, PPLKEY):
            return self.key == other.key and self.val == other.val
        return False

    def __hash__(self):
        return hash((self.key, self.val))

class MapCounter:
    def __init__(self, reverse=False):
        self.map = {}
        self.reverse = reverse
        self.sorted_keys = []

    def add(self, key, cnt=1):
        if key in self.map:
            self.map[key] += cnt
        else:
            self.map[key] = cnt
        if not self.sorted_keys or (self.reverse and key < self.sorted_keys[-1]) or (not self.reverse and key > self.sorted_keys[-1]):
            self.sorted_keys.append(key)
        else:
            for i in range(len(self.sorted_keys)):
                if (self.reverse and key < self.sorted_keys[i]) or (not self.reverse and key > self.sorted_keys[i]):
                    self.sorted_keys.insert(i, key)
                    break
            else:
                self.sorted_keys.append(key)

    def remove(self, key):
        self.sub(key, 1, False)

    def sub(self, key, cnt=1, minus=True):
        if key in self.map:
            new_val = self.map[key] - cnt
            if new_val == 0 or (not minus and new_val < 0):
                del self.map[key]
                self.sorted_keys.remove(key)
            else:
                self.map[key] = new_val

    def getCount(self, key):
        return self.map.get(key, 0)

    def getKeyCount(self):
        return len(self.map)

    def getFirstKey(self):
        return self.sorted_keys[0] if self.sorted_keys else None

    def getLastKey(self):
        return self.sorted_keys[-1] if self.sorted_keys else None

    def clear(self):
        self.map.clear()
        self.sorted_keys.clear()

def isRightMin(a, f, index, key):
    if f and a[index] >= key:
        return True
    elif not f and a[index] > key:
        return True
    else:
        return False

def binarySearchRightMin(a, f, key):
    ng = -1
    ok = len(a)
    while abs(ok - ng) > 1:
        mid = (ok + ng) // 2
        if isRightMin(a, f, mid, key):
            ok = mid
        else:
            ng = mid
    return ok

def isLeftMax(a, f, index, key):
    if f and a[index] <= key:
        return True
    elif not f and a[index] < key:
        return True
    else:
        return False

def binarySearchLeftMax(a, f, key):
    ng = -1
    ok = len(a)
    while abs(ok - ng) > 1:
        mid = (ok + ng) // 2
        if isLeftMax(a, f, mid, key):
            ng = mid
        else:
            ok = mid
    return ng

class EulerTour:
    def __init__(self):
        self.g = None
        self.euler_tour = []
        self.begin = None
        self.end = None
        self.k = 0
        self.root = 0

    def dfs(self, v, p, out):
        out.write(f"v = {v}  p = {p}\n")
        self.begin[v] = self.k
        self.euler_tour.append(v)
        self.k += 1
        if v not in self.g.data:
            return
        for i in self.g.data[v]:
            if i != p:
                self.dfs(i, v, out)
                self.euler_tour.append(v)
                self.k += 1
        self.end[v] = self.k

    def init(self, p_cnt, root, g, out):
        self.begin = [0] * (p_cnt + 1)
        self.end = [0] * (p_cnt + 1)
        self.root = root
        self.g = g
        self.k = 0
        self.euler_tour = []
        self.dfs(root, -1, out)

    def getPartTour(self, v):
        return self.euler_tour[self.begin[v]: self.end[v]]

    def getPartList(self, v):
        s = set(self.getPartTour(v))
        return list(s)

class Graph:
    def __init__(self):
        self.data = {}
        self.mapCnt = {}
        self.mapLow = {}
        self.mark = set()
        self.number = 0

    def add(self, from_node, to_node):
        if from_node not in self.data:
            self.data[from_node] = []
        self.data[from_node].append(to_node)

    def del_edge(self, from_node, to_node):
        if from_node in self.data:
            if to_node in self.data[from_node]:
                self.data[from_node].remove(to_node)

    def get(self, key):
        return self.data.get(key, [])

    def contains(self, key):
        return key in self.data

    def keySet(self):
        return list(self.data.keys())

    def isConnect(self, key1, key2):
        return key2 in self.data.get(key1, [])

    def distList(self, key):
        dist = []
        mark = set()
        stack = []
        stack.append((key, 0))
        while stack:
            (pp, dd) = stack.pop()
            if pp in mark:
                continue
            mark.add(pp)
            dist.append(PP(pp, dd))
            for next_node in self.data.get(pp, []):
                if next_node not in mark:
                    stack.append((next_node, dd + 1))
        return dist

    def distV(self, key):
        dist = {}
        for k in self.data.keys():
            dist[k] = -1
        stack = []
        stack.append((key, 0))
        dist[key] = 0
        while stack:
            (pp, dd) = stack.pop()
            for next_node in self.data.get(pp, []):
                if dist[next_node] == -1:
                    dist[next_node] = dd + 1
                    stack.append((next_node, dd + 1))
        return dist

    def bridgeDfs(self, now, pre):
        self.mark.add(now)
        self.mapCnt[now] = self.number
        self.mapLow[now] = self.number
        self.number += 1
        low = self.mapLow[now]
        for next_node in self.data.get(now, []):
            if next_node == pre:
                continue
            if next_node in self.mark:
                if self.mapLow[now] > self.mapLow[next_node]:
                    self.mapLow[now] = self.mapLow[next_node]
                continue
            current_low = self.bridgeDfs(next_node, now)
            if self.mapLow[now] > current_low:
                self.mapLow[now] = current_low
        return self.mapLow[now]

    def bridgeCnt(self, start):
        self.mapCnt.clear()
        self.mapLow.clear()
        self.mark.clear()
        self.number = 0
        self.bridgeDfs(start, start)
        ans = 0
        for key in self.mapCnt:
            if self.mapCnt[key] == self.mapLow[key]:
                ans += 1
        return ans - 1

    def dump(self, out):
        for key in self.data:
            out.write(f"{key} : ")
            for val in self.data[key]:
                out.write(f"{val} ")
            out.write("\n")

class GraphWith:
    def __init__(self):
        self.data = {}

    def add(self, key, pp):
        if key not in self.data:
            self.data[key] = []
        self.data[key].append(pp)

    def get(self, key):
        return self.data.get(key, [])

    def contains(self, key):
        return key in self.data

    def keySet(self):
        return list(self.data.keys())

    def isConnect(self, key1, key2):
        if key1 not in self.data:
            return False
        for p in self.data[key1]:
            if p.key == key2:
                return True
        return False

class GraphLong:
    def __init__(self):
        self.G = {}

    def add(self, key, value):
        if key not in self.G:
            self.G[key] = []
        self.G[key].append(value)

    def get(self, key):
        return self.G.get(key, [])

class GraphLongWith:
    def __init__(self):
        self.G = {}

    def add(self, key, ppl):
        if key not in self.G:
            self.G[key] = []
        self.G[key].append(ppl)

    def get(self, key):
        return self.G.get(key, [])

class Solver:
    def __init__(self, in_reader, out):
        self.in_reader = in_reader
        self.out = out

    def solve(self):
        H = self.in_reader.nextInt()
        W = self.in_reader.nextInt()
        C = [['.' for _ in range(W + 2)] for _ in range(H + 2)]
        for i in range(1, H + 1):
            line = self.in_reader.nextString().strip()
            for j in range(1, W + 1):
                C[i][j] = line[j - 1]
        if H == 1 and W == 1:
            self.out.write("No\n")
            self.out.flush()
            return
        for i in range(1, H + 1):
            for j in range(1, W + 1):
                if C[i][j] == '#' and C[i-1][j] != '#' and C[i+1][j] != '#' and C[i][j-1] != '#' and C[i][j+1] != '#':
                    self.out.write("No\n")
                    self.out.flush()
                    return
        self.out.write("Yes\n")
        self.out.flush()

def main():
    in_reader = MyInput()
    out = sys.stdout
    solver = Solver(in_reader, out)
    solver.solve()
    out.flush()

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/multiline/uncomment/direct/Java/C++/s893190600_commented.cpp
#include <iostream>
#include <vector>
#include <string>
#include <map>
#include <set>
#include <queue>
#include <stack>
#include <algorithm>
#include <cmath>
#include <sstream>
#include <iomanip>

using namespace std;

class MyInput {
private:
    istream& in;
    string buffer;
    size_t pos;

public:
    MyInput(istream& is) : in(is), pos(0) {}

    int read() {
        if (pos >= buffer.size()) {
            getline(in, buffer);
            pos = 0;
        }
        return pos < buffer.size() ? buffer[pos++] : -1;
    }

    int nextInt() {
        int c = skipSpaces();
        bool negative = c == '-';
        if (negative) c = read();
        int result = 0;
        while (isdigit(c)) {
            result = result * 10 + (c - '0');
            c = read();
        }
        return negative ? -result : result;
    }

    long long nextLong() {
        int c = skipSpaces();
        bool negative = c == '-';
        if (negative) c = read();
        long long result = 0;
        while (isdigit(c)) {
            result = result * 10 + (c - '0');
            c = read();
        }
        return negative ? -result : result;
    }

    double nextDouble() {
        int c = skipSpaces();
        bool negative = c == '-';
        if (negative) c = read();
        double result = 0;
        while (isdigit(c)) {
            result = result * 10 + (c - '0');
            c = read();
        }
        if (c == '.') {
            c = read();
            double fraction = 0.1;
            while (isdigit(c)) {
                result += fraction * (c - '0');
                fraction *= 0.1;
                c = read();
            }
        }
        return negative ? -result : result;
    }

    string nextString() {
        int c = skipSpaces();
        string result;
        while (!isspace(c) && c != -1) {
            result += (char)c;
            c = read();
        }
        return result;
    }

private:
    int skipSpaces() {
        int c;
        do {
            c = read();
        } while (isspace(c) && c != -1);
        return c;
    }
};

struct PP {
    int key, val;
    PP(int k, int v) : key(k), val(v) {}
};

struct PPP {
    int key, val1, val2;
    PPP(int k, int v1, int v2) : key(k), val1(v1), val2(v2) {}
};

struct PPL {
    long long key, val;
    PPL(long long k, long long v) : key(k), val(v) {}
};

struct PPDL {
    long long key;
    vector<long long> val;
    PPDL(long long k, vector<long long> v) : key(k), val(v) {}
    void dump(ostream& out) {
        out << "key = " << key << "  val ";
        for (long long v : val) {
            out << "[" << v << "] ";
        }
        out << endl;
    }
};

class MapCounter {
private:
    map<int, int> map;

public:
    MapCounter() {}

    void add(int key) {
        add(key, 1);
    }

    void add(int key, int cnt) {
        map[key] += cnt;
    }

    void remove(int key) {
        sub(key, 1, false);
    }

    void sub(int key) {
        sub(key, 1);
    }

    void sub(int key, int cnt) {
        sub(key, cnt, true);
    }

    void sub(int key, int cnt, bool minus) {
        if (map.find(key) != map.end()) {
            if (map[key] > cnt || minus) {
                map[key] -= cnt;
                if (map[key] == 0) {
                    map.erase(key);
                }
            }
        } else if (minus) {
            map[key] = -cnt;
        }
    }

    void set(int key, int cnt) {
        map[key] = cnt;
    }

    int getCount(int key) {
        return map.count(key) ? map[key] : 0;
    }

    set<int> getKey() {
        set<int> keys;
        for (const auto& pair : map) {
            keys.insert(pair.first);
        }
        return keys;
    }

    int getKeyCount() {
        return map.size();
    }

    int getFirstKey() {
        return map.begin()->first;
    }

    int getLastKey() {
        return map.rbegin()->first;
    }

    void clear() {
        map.clear();
    }
};

class Graph {
private:
    map<int, vector<int>> data;

public:
    void add(int from, int to) {
        data[from].push_back(to);
    }

    void del(int from, int to) {
        if (data.find(from) != data.end()) {
            data[from].erase(remove(data[from].begin(), data[from].end(), to), data[from].end());
        }
    }

    vector<int> get(int key) {
        return data.count(key) ? data[key] : vector<int>();
    }

    bool contains(int key) {
        return data.count(key);
    }

    set<int> keySet() {
        set<int> keys;
        for (const auto& pair : data) {
            keys.insert(pair.first);
        }
        return keys;
    }

    bool isConnect(int key_1, int key_2) {
        if (data.find(key_1) != data.end()) {
            return find(data[key_1].begin(), data[key_1].end(), key_2) != data[key_1].end();
        }
        return false;
    }

    vector<PP> distList(int key) {
        vector<PP> dist;
        set<int> mark;
        stack<PP> stack;
        stack.push(PP(key, 0));
        while (!stack.empty()) {
            PP wk = stack.top();
            stack.pop();
            int pp = wk.key;
            int dd = wk.val;
            if (mark.count(pp)) continue;
            mark.insert(pp);
            dist.push_back(wk);
            for (int next : get(pp)) {
                if (mark.count(next)) continue;
                stack.push(PP(next, dd + 1));
            }
        }
        return dist;
    }

    int* distV(int key) {
        int* dist = new int[data.size() + 1];
        fill(dist, dist + data.size() + 1, -1);
        set<int> mark;
        stack<PP> stack;
        stack.push(PP(key, 0));
        while (!stack.empty()) {
            PP wk = stack.top();
            stack.pop();
            int pp = wk.key;
            int dd = wk.val;
            if (mark.count(pp)) continue;
            mark.insert(pp);
            dist[pp] = dd;
            for (int next : get(pp)) {
                if (mark.count(next)) continue;
                stack.push(PP(next, dd + 1));
            }
        }
        return dist;
    }

    map<int, int> mapCnt;
    map<int, int> mapLow;
    set<int> mark;
    int number;

    int bridgeDfs(int now, int pre) {
        mark.insert(now);
        mapCnt[now] = number;
        mapLow[now] = number;
        int low;
        for (int next : get(now)) {
            if (next == pre) continue;
            if (mark.count(next)) {
                if (mapLow[now] > mapLow[next]) {
                    mapLow[now] = mapLow[next];
                }
                continue;
            }
            number++;
            low = bridgeDfs(next, now);
            if (mapLow[now] > low) {
                mapLow[now] = low;
            }
        }
        return mapLow[now];
    }

    int bridgeCnt(int start) {
        mapCnt.clear();
        mapLow.clear();
        mark.clear();
        number = 0;
        bridgeDfs(start, start);
        int ans = 0;
        for (const auto& pair : mapCnt) {
            if (mapCnt[pair.first] == mapLow[pair.first]) {
                ans++;
            }
        }
        return ans - 1;
    }

    void dump(ostream& out) {
        for (const auto& pair : data) {
            out << pair.first << " : ";
            for (int val : pair.second) {
                out << val << " ";
            }
            out << endl;
        }
    }
};

class GraphWith {
private:
    map<int, vector<PP>> data;

public:
    void add(int key, PP p) {
        data[key].push_back(p);
    }

    vector<PP> get(int key) {
        return data.count(key) ? data[key] : vector<PP>();
    }

    bool contains(int key) {
        return data.count(key);
    }

    set<int> keySet() {
        set<int> keys;
        for (const auto& pair : data) {
            keys.insert(pair.first);
        }
        return keys;
    }

    bool isConnect(int key_1, int key_2) {
        if (data.find(key_1) != data.end()) {
            for (const PP& p : data[key_1]) {
                if (p.key == key_2) return true;
            }
        }
        return false;
    }

    int distance(int key_1, int key_2) {
        set<int> mark;
        stack<PP> stack;
        stack.push(PP(key_1, 0));
        while (!stack.empty()) {
            PP wk = stack.top();
            stack.pop();
            int key = wk.key;
            int val = wk.val;
            if (mark.count(key)) continue;
            mark.insert(key);
            if (key == key_2) return val;
            for (const PP& pp : get(key)) {
                if (mark.count(pp.key)) continue;
                stack.push(PP(pp.key, val + pp.val));
            }
        }
        return INT_MAX;
    }
};

class GraphLong {
private:
    map<long long, vector<long long>> G;

public:
    void add(long long key, long long value) {
        G[key].push_back(value);
    }

    vector<long long> get(long long key) {
        return G.count(key) ? G[key] : vector<long long>();
    }
};

class GraphLongWith {
private:
    map<long long, vector<PPL>> G;

public:
    void add(long long key, PPL p) {
        G[key].push_back(p);
    }

    vector<PPL> get(long long key) {
        return G.count(key) ? G[key] : vector<PPL>();
    }
};

class Solver {
private:
    MyInput& in;
    ostream& out;

public:
    Solver(MyInput& in, ostream& out) : in(in), out(out) {}

    void solve() {
        int H = in.nextInt();
        int W = in.nextInt();
        vector<vector<char>> C(H + 2, vector<char>(W + 2));
        for (int i = 0; i < H; i++) {
            string line = in.nextString();
            for (int j = 0; j < W; j++) {
                C[i + 1][j + 1] = line[j];
            }
        }
        if (H == 1 && W == 1) {
            out << "No" << endl;
            return;
        }
        for (int i = 1; i <= H; i++) {
            for (int j = 1; j <= W; j++) {
                if (C[i][j] == '#' && C[i - 1][j] != '#' && C[i + 1][j] != '#' &&
                    C[i][j - 1] != '#' && C[i][j + 1] != '#') {
                    out << "No" << endl;
                    return;
                }
            }
        }
        out << "Yes" << endl;
    }
};

int main() {
    MyInput in(cin);
    Solver solver(in, cout);
    solver.solve();
    return 0;
}
// End of Code
